src/bun.js/api/bun/process.zig

```diff
@@ -1716,6 +1716,13 @@ pub fn spawnProcessWindows(
         .extra_pipes = try std.ArrayList(WindowsSpawnResult.StdioResult).initCapacity(bun.default_allocator, options.extra_fds.len),
     };
 
+    // By putting this process in a Job Object, any child process will be caught
+    // so that killing this process will kill all children processes.
+    //
+    // It seems that just associating the process does the work neccecary.
+    if (bun.windows.CreateJobObjectA(null, null)) |job|
+        _ = bun.windows.AssignProcessToJobObject(job, process.poller.uv.process_handle);
```

Comment:
What happens if the process has already spawned a child process and then exited by the time it's being assigned to the job object? See [this blog](https://devblogs.microsoft.com/oldnewthing/20230209-00/?p=107812) for context. Not sure if this is an edge case we should care about, but might be relevant.

To do it the "right" way we might need to mess with libuv internals for process spawning, not sure if we want to do that.

Reply:
i moved this to just the code for --watch (libuv already does this and actually did it more correct than what i have here), and i followed the linked guide so that we spawn the process without this race. thank you.

Reply:
Looks good now 👍